### DISCUSSION TITLE
barrel file results in huge import of everything

### DISCUSSION THREAD

**Discussion by dagda1** [themesberg/flowbite-react#1330] [Ideas]
Unfortunately when I have the following import

```ts
import { Modal } from 'flowbite-react';
```

This will pull in absolutely everything from the [barrel file](https://github.com/themesberg/flowbite-react/blob/main/packages/ui/src/index.ts).

What would be better is to have the ability just to grab the modal:

```ts
import Modal from 'flowbite-react/Modal';
```

This could be achieved with package.json exports, something like

```json
{
  "name": "flowbite-react",
  "exports": {
    "./Modal": "./wherever/Modal/index.js",
    // other components...
  }
}

```

**Comment by MiroslavPetrik** [comment]
This was fixed here https://github.com/themesberg/flowbite-react/issues/1197

I just tested it here https://github.com/MiroslavPetrik/net3-starter/commit/b2d7205467ae4f5595ab4e69f1528695fe0aeff8.

Took me a while to get to the path `flowbite-react/components/`, as its not in documentation as it uses barrel export everywhere: 
https://flowbite-react.com/docs/components/button

it reduced build size by over 40kb

![exact-import](https://github.com/user-attachments/assets/2a7e7009-dc80-4a30-8010-83211832789b)
<img width="485" alt="barrel-file" src="https://github.com/user-attachments/assets/4479e3cf-e3ef-4ca1-ad89-9bbb31cc01e1">
